### PR TITLE
luccaUI/tablegrid [ADD] possibility to display a "checkbox" column as…

### PR DESCRIPTION
**Pull request by srabere**
… first column.

### CODE REVIEW

**Comment by lucienbertin** [comment]
update the changelog and add unit tests or e2e tests

File: ts/table-grid/table-grid.directive.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ module Lui.Directives {
 		public static IID = "luidTableGrid";
 		public controller = "luidTableGridController";
 		public restrict = "AE";
-		public scope = { header: "=", height: "@", datas: "=" };
+		public scope = { header: "=", height: "@", datas: "=", hasCheckbox: "@" };
```
Comment (PaulAndreLOUIS) [inline ts/table-grid/table-grid.directive.ts]: hasCheckboxes ?

File: ts/table-grid/table-grid.directive.ts [hunk 2]
```diff
@@ -45,7 +45,8 @@ module Lui.Directives {
 					let scrollableHeader: any = datagridHeader.querySelector(".columns:not(.locked)"); // right part of header
 					let scrollableTable: any = scrollableContent.querySelector("table"); // right part of the rows
 
-					let lockedWidth = _.reduce(scope.fixedRowDefinition, (memo: number, num: TableGrid.Header) => { return memo + num.width; }, 0) + "em";
+					let indexIteration = scope.hasCheckbox ? 3 : 0;
```
Comment (PaulAndreLOUIS) [inline ts/table-grid/table-grid.directive.ts]: magic number 3 ?

File: ts/table-grid/table-grid.controller.ts [hunk 3]
```diff
@@ -64,6 +64,7 @@ module Lui.Directives {
 				$scope.fixedRowDefinition = [];
 				$scope.scrollableHeaderRows = [];
 				$scope.scrollableRowDefinition = [];
+				$scope.allChecked = {value: false};
```
Comment (PaulAndreLOUIS) [inline ts/table-grid/table-grid.controller.ts]: why an object just to store a boolean ?

File: ts/table-grid/table-grid.controller.ts [hunk 4]
```diff
@@ -150,6 +151,10 @@ module Lui.Directives {
 				return tmp.textContent || tmp.innerText || "";
 			};
 
+			$scope.updateCheckboxes = () => {
+				_.each($scope.datas, (data: any) => { data.isChecked = $scope.allChecked.value; });
+			};
```
Comment (PaulAndreLOUIS) [inline ts/table-grid/table-grid.controller.ts]: maybe something like 
```
$scope.onMasterCheckBoxesChange: () => {
    if (_.some(rows, (row) => { !row.isChecked; })
    {
        _.each(rows, (row) => { row.isChecked = true; })
    } else {
        _.each(rows, (row) => { row.isChecked = false; })
    }
}
```

You don't need $scope.allChecked anymore plus you can uncked everything with a double click on the master checkbox.

Comment (srabere) [inline ts/table-grid/table-grid.controller.ts]: It is a good idea but something is missing. I have to add a method on each checkbox to unselect master checkbox if necessary.

File: ts/table-grid/table-grid.controller.ts (hunk 3 above)
Comment (srabere) [inline ts/table-grid/table-grid.controller.ts]: required to use data-binding with checkboxes : http://stackoverflow.com/questions/18642371/checkbox-not-binding-to-scope-in-angularjs.
But, this property will be removed using your solution below.

**Pull request closed without merging**